File: javascript/readFile.js
#!/usr/bin/env node

/**
 * Usage: ./readFile.js [filepath]
 */

const fs = require("fs");

const params = process.argv.slice(2);

if (params.length > 0) {
    const filepath = params[0];
    const lines = fs.readFileSync(filepath, "utf8").split("\n");
    processFileLines(lines);
} else {
    console.error("Missing file argument.");
}

function processFileLines(lines) {
    console.log("Printing file lines:")
    lines.forEach((line) => {
        // TODO: implementation
        console.log(line)
    });
}

File: python/read_file.py
#!/usr/bin/env python

# Usage: ./read_line.py [filepath]

def read_file(filepath):
    print "Printing file lines:"
    with open(filepath) as f:
        for line in f:
            # TODO: implementation
            print line,

if __name__ == "__main__":
    import sys
    if len(sys.argv) > 1:
        read_file(sys.argv[1])
    else:
        print "Missing file argument."
